internal/logger/target/http/http.go

```diff
@@ -275,90 +254,244 @@ func (h *Target) send(ctx context.Context, payload []byte, payloadType string, t
 	}
 }
 
-func (h *Target) logEntry(ctx context.Context, payload []byte, payloadType string) {
-	const maxTries = 3
-	tries := 0
-	for tries < maxTries {
-		if atomic.LoadInt32(&h.status) == statusClosed {
-			// Don't retry when closing...
-			return
-		}
-		// sleep = (tries+2) ^ 2 milliseconds.
-		sleep := time.Duration(math.Pow(float64(tries+2), 2)) * time.Millisecond
-		if sleep > time.Second {
-			sleep = time.Second
-		}
-		time.Sleep(sleep)
-		tries++
-		err := h.send(ctx, payload, payloadType, webhookCallTimeout)
-		if err == nil {
-			return
-		}
-		h.config.LogOnce(ctx, err, h.Endpoint())
-	}
-	if tries == maxTries {
-		// Even with multiple retries, count failed messages as only one.
-		atomic.AddInt64(&h.failedMessages, 1)
+func (h *Target) startQueueProcessor(ctx context.Context, mainWorker bool) {
+	h.logChMu.RLock()
+	if h.logCh == nil {
+		h.logChMu.RUnlock()
+		return
 	}
-}
+	h.logChMu.RUnlock()
 
-func (h *Target) startHTTPLogger(ctx context.Context) {
 	atomic.AddInt64(&h.workers, 1)
 	defer atomic.AddInt64(&h.workers, -1)
 
-	h.logChMu.RLock()
-	logCh := h.logCh
-	if logCh != nil {
-		// We are not allowed to add when logCh is nil
-		h.wg.Add(1)
-		defer h.wg.Done()
-	}
-	h.logChMu.RUnlock()
-	if logCh == nil {
-		return
-	}
+	h.wg.Add(1)
+	defer h.wg.Done()
+
+	entries := make([]interface{}, 0)
+	name := h.Name()
+
+	defer func() {
+		// re-load the global buffer pointer
+		// in case it was modified by a new target.
+		logChLock.Lock()
+		currentGlobalBuffer, ok := logChBuffers[name]
+		logChLock.Unlock()
+		if !ok {
+			return
+		}
+
+		for _, v := range entries {
+			select {
+			case currentGlobalBuffer <- v:
+			default:
+			}
+		}
+
+		if mainWorker {
+		drain:
+			for {
+				select {
+				case v, ok := <-h.logCh:
+					if !ok {
+						break drain
+					}
+
+					currentGlobalBuffer <- v
```

Comment:
If we are looking for some in-memory batch, we have got similar implementation in https://github.com/minio/minio/blob/master/internal/event/target/kafka.go#L161 (kafka bucket notification target). Looks like it is a generic already, we can use it If you think it helps.